docs/enhancements/aws-private-link.md

```diff
@@ -0,0 +1,553 @@
+```
+route53.ChangeResourceRecordSets
+  Id: "{hosted zone Id}",
+  "ChangeBatch": {
+    "Comment": "Add TXT record for domain verification",
+    "Changes": [
+      {
+        "Action": "UPSERT",
+        "ResourceRecordSet": {
+          "Name": "{DomainName}.{apiOverrideURL}",
+          "Type": "TXT",
+          "RecordSets": [{
+            "Value": "{DomainValue}"
+          }]
+        }
+      }
+    ]
+  }
+}
+```
+
+3. Run manual verification for the domain
+Afer the route53 zone was updated, the controller needs to manually trigger the
+verification of the domain.
+
+https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_StartVpcEndpointServicePrivateDnsVerification.html
+```
+ec2.StartVpcEndpointServicePrivateDnsVerification
+  ServiceId: "{serviceId}"
+```
+
+4. Wait for verification to complete
+
+Wait for the verification to be successful, by making sure the `state` is `verified`
+in the PrivateDnsConfiguration for the endpoint service.
+
+#### Creating VPC Endpoint
+
+For the controller to create the VPC endpoints it needs to compute the VPC and
+the subnets.
+
+The subnets that should be used depend on the AZs that are available for the endpoint
+service. To figure out which AZs are supported by the service, the controller
+needs to describe the endpoint service from the context of the hub account.
+
+https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcEndpointServices.html
+```
+ec2.DescribeVpcEndpointServices
+  ServiceName:
+  - "{serviceName}"
+```
+
+The response includes a list of avaialabilty zones. These are the friendly names
+for the AZs in hub account.
+NOTE: this list can be different from the cluster account as friendly names are
+different in different accounts.
+
+The controller filters the list of subnets using the AZs. Using subnets in hub
+account for all available AZs allows maximum availabity of connection.
+
+The controller can then create VPC endpoint using
+https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpcEndpoint.html
+```
+ec2.CreateVpcEndpoint
+  PrivateDnsEnabled: True
+  ServiceName: "{serviceName}"
+  SubnetId:
+  - subnet-1
+  - subnet-2 or more
+  VpcId: "{computed VPC id from subnets}"
+  VpcEndpointType: Interface
+```
+
+The controller stores the VPC endpoint Id to the ClusterDeployment.
+
+#### Waiting for VPC Endpoint approval
+
+The endpoint service is allowed to automatically approve any endpoint created by
+the allowed IAM principals. So the controller doesn't need to take additional
+steps but since approval can take some time, it needs to wait for approval.
+
+The controller can wait for approval using
+https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcEndpoints.html
+```
+ec2.DescribeVpcEndpoints
+  VpcEndpointId:
+  - "{endpoint Id}"
+```
+
+The controller ensures that the state of the VPC endpoint is `Available`.
+
+#### Failed installations
```

Comment:
Also worth fleshing out a deleting clusters section to understand what operations are performed outside of the installer and ensure all steps are understood.

Reply:
added `#### Deprovisioning PrivateLink` section for that.